### ISSUE TITLE
Can't locate object method "NAME" via package "B::SPECIAL" at /opt/perl/lib/perl5/B/Concise.pm line 714.

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#13436]
Migrated from [rt.perl.org#120609](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=120609) (status was 'resolved')

Searchable as RT120609$

**Comment by p5pRT** [comment]
### From schmorp@schmorp.de

#### Created by schmorp@schmorp.de



\-MO=Terse has problems with the following snippet&#8203;:

&nbsp;   \# perl \-MO=Terse \-e 'print \<\\n>'

It prints&#8203;:

&nbsp;   LISTOP \(0x5f6788\) leave \[1\] 
&nbsp;       OP \(0x5f6750\) enter 
&nbsp;       COP \(0x5f67d0\) nextstate 
&nbsp;       LISTOP \(0x5ba1c0\) print 
&nbsp;           OP \(0x5f6830\) pushmark 
&nbsp;           LISTOP \(0x5ba310\) glob \[1\] 
&nbsp;               OP \(0x5ba2d8\) null \[3\] 
&nbsp;               SVOP \(0x5ba358\) const  PV \(0x5840e0\) "\\n" 
&nbsp;   Can't locate object method "NAME" via package "B&#8203;::SPECIAL" at /opt/perl/lib/perl5/B/Concise\.pm line 714\.
&nbsp;   CHECK failed\-\-call queue aborted\.

<details><summary>Perl Info</summary>

```

Flags:
    category=library
    severity=low
    module=O

Site configuration information for perl 5.18.1:

Configured by Marc Lehmann at Mon Oct 21 09:27:08 CEST 2013.

Summary of my perl5 (revision 5 version 18 subversion 1) configuration:
   
  Platform:
    osname=linux, osvers=3.10-3-amd64, archname=x86_64-linux
    uname='linux cerebro 3.10-3-amd64 #1 smp debian 3.10.11-1 (2013-09-10) x86_64 gnulinux '
    config_args='-Duselargefiles -Duse64bitint -Dusemymalloc=n -Dstatic_ext=Fcntl -Dcc=ccache gcc -Dccflags=-DPERL_DISABLE_PMC -DPERL_ARENA_SIZE=16376 -USITEARCH_EXP -USITELIB_EXP -UARCHLIB_EXP -D_GNU_SOURCE  -I/opt/include -ggdb -gdwarf-2 -g3 -Doptimize=-O6 -fno-asynchronous-unwind-tables -fno-strict-aliasing -Dcccdlflags=-fPIC -Dldflags=-L/opt/perl/lib -L/opt/lib -Dlibs=-ldl -lm -lcrypt -Dprefix=/opt/perl -Dprivlib=/opt/perl/lib/perl5 -Darchlib=/opt/perl/lib/perl5 -Uusevendorprefix -Dsiteprefix=/opt/perl -Dsitelib=/opt/perl/lib/perl5 -Dsitearch=/opt/perl/lib/perl5 -Dsitebin=/opt/perl/bin -Dman1dir=/opt/perl/man/man1 -Dman3dir=/opt/perl/man/man3 -Dsiteman1dir=/opt/perl/man/man1 -Dsiteman3dir=/opt/perl/man/man3 -Dman1ext=1 -Dman3ext=3 -Dpager=/usr/bin/less -Uafs -Uusesfio -Uusenm -Uuseshrplib -Ud_dosuid -Dusethreads=undef -Duse5005threads=undef -Duseithreads=undef -Dusemultiplicity=undef -Demail=perl-binary@plan9.de -Dcf_email=perl-binary@plan9.de -Dcf_by=Marc Lehmann -Dlocincpth=/opt/perl/include /opt/include -Dmyhostname=localhost -Dmultiarch=undef -Dbin=/opt/perl/bin -Dxxxusedevel -DxxxDEBUGGING -Dxxxuse_debugging_perl -Dxxxuse_debugmalloc -dEs'
    hint=recommended, useposix=true, d_sigaction=define
    useithreads=undef, usemultiplicity=undef
    useperlio=define, d_sfio=undef, uselargefiles=define, usesocks=undef
    use64bitint=define, use64bitall=define, uselongdouble=undef
    usemymalloc=n, bincompat5005=undef
  Compiler:
    cc='ccache gcc', ccflags ='-DPERL_DISABLE_PMC -DPERL_ARENA_SIZE=16376 -USITEARCH_EXP -USITELIB_EXP -UARCHLIB_EXP -D_GNU_SOURCE -I/opt/include -ggdb -gdwarf-2 -g3 -fno-strict-aliasing -pipe -I/opt/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64',
    optimize='-O6 -fno-asynchronous-unwind-tables -fno-strict-aliasing',
    cppflags='-DPERL_DISABLE_PMC -DPERL_ARENA_SIZE=16376 -USITEARCH_EXP -USITELIB_EXP -UARCHLIB_EXP -D_GNU_SOURCE -I/opt/include -ggdb -gdwarf-2 -g3 -fno-strict-aliasing -pipe -I/opt/include'
    ccversion='', gccversion='4.7.2', gccosandvers=''
    intsize=4, longsize=8, ptrsize=8, doublesize=8, byteorder=12345678
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=16
    ivtype='long', ivsize=8, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=8
    alignbytes=8, prototype=define
  Linker and Libraries:
    ld='ccache gcc', ldflags ='-L/opt/perl/lib -L/opt/lib -L/usr/local/lib'
    libpth=/usr/local/lib /lib/x86_64-linux-gnu /lib/../lib /usr/lib/x86_64-linux-gnu /usr/lib/../lib /lib /usr/lib
    libs=-ldl -lm -lcrypt
    perllibs=-ldl -lm -lcrypt
    libc=, so=so, useshrplib=false, libperl=libperl.a
    gnulibc_version='2.17'
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags='-Wl,-E'
    cccdlflags='-fPIC', lddlflags='-shared -O6 -fno-asynchronous-unwind-tables -fno-strict-aliasing -L/opt/perl/lib -L/opt/lib -L/usr/local/lib'

Locally applied patches:
    


@INC for perl 5.18.1:
    /root/src/sex
    /root/pserv/lib/perl5
    /opt/perl/lib/perl5
    /opt/perl/lib/perl5
    .


Environment for perl 5.18.1:
    HOME=/root
    LANG (unset)
    LANGUAGE (unset)
    LC_CTYPE=en_US.UTF-8
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/root/s2:/root/s:/opt/bin:/opt/sbin:/bin:/sbin:/usr/bin:/usr/sbin:/usr/X11/bin:/usr/games:/usr/local/bin:/usr/local/sbin:/root/pserv:.
    PERL5LIB=/root/src/sex:/root/pserv/lib/perl5
    PERL5_CPANPLUS_CONFIG=/root/.cpanplus/config
    PERLDB_OPTS=ornaments=0
    PERL_ANYEVENT_DBI_TESTS=1
    PERL_ANYEVENT_LOOP_TESTS=1
    PERL_ANYEVENT_NET_TESTS=1
    PERL_BADLANG (unset)
    PERL_UNICODE=E
    SHELL=/bin/bash

```
</details>

**Comment by p5pRT** [comment]
### From tsibley@cpan.org

On 11/21/2013 10&#8203;:01 PM\, schmorp@&#8203;schmorp\.de \(via RT\) wrote&#8203;:
> \-MO=Terse has problems with the following snippet&#8203;:
> 
>    \# perl \-MO=Terse \-e 'print \<\\n>'
> 
> It prints&#8203;:
> 
>    LISTOP \(0x5f6788\) leave \[1\] 
>        OP \(0x5f6750\) enter 
>        COP \(0x5f67d0\) nextstate 
>        LISTOP \(0x5ba1c0\) print 
>            OP \(0x5f6830\) pushmark 
>            LISTOP \(0x5ba310\) glob \[1\] 
>                OP \(0x5ba2d8\) null \[3\] 
>                SVOP \(0x5ba358\) const  PV \(0x5840e0\) "\\n" 
>    Can't locate object method "NAME" via package "B&#8203;::SPECIAL" at /opt/perl/lib/perl5/B/Concise\.pm line 714\.
>    CHECK failed\-\-call queue aborted\.

I believe this was fixed by 7b69624\, first released in 5\.19\.4\.

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From @jkeenan

On Thu Nov 21 22&#8203;:28&#8203;:56 2013\, tsibley wrote&#8203;:
> On 11/21/2013 10&#8203;:01 PM\, schmorp@&#8203;schmorp\.de \(via RT\) wrote&#8203;:
> > \-MO=Terse has problems with the following snippet&#8203;:
> >
> > \# perl \-MO=Terse \-e 'print \<\\n>'
> >
> > It prints&#8203;:
> >
> > LISTOP \(0x5f6788\) leave \[1\]
> >     OP \(0x5f6750\) enter
> >     COP \(0x5f67d0\) nextstate
> >     LISTOP \(0x5ba1c0\) print
> >         OP \(0x5f6830\) pushmark
> >         LISTOP \(0x5ba310\) glob \[1\]
> >             OP \(0x5ba2d8\) null \[3\]
> >             SVOP \(0x5ba358\) const  PV \(0x5840e0\) "\\n"
> > Can't locate object method "NAME" via package "B&#8203;::SPECIAL" at
> > /opt/perl/lib/perl5/B/Concise\.pm line 714\.
> > CHECK failed\-\-call queue aborted\.
> 
> I believe this was fixed by 7b69624\, first released in 5\.19\.4\.

Here is confirmation\, from blead&#8203;:

\#\#\#\#\#
$ \./perl \-Ilib \-v

This is perl 5\, version 19\, subversion 7 \(v5\.19\.7 \(v5\.19\.6\-26\-ga2235d7\)\) built for x86\_64\-linux
\.\.\.
\#\#\#\#\#
$ \./perl \-Ilib \-MO=Terse \-e 'print \<\\n>'
LISTOP \(0x1d35798\) leave \[1\] 
&nbsp;    OP \(0x1d35760\) enter 
&nbsp;    COP \(0x1d357e0\) nextstate 
&nbsp;    LISTOP \(0x1cdb430\) print 
&nbsp;        OP \(0x1d35840\) pushmark 
&nbsp;        LISTOP \(0x1cdb580\) glob \[1\] 
&nbsp;            OP \(0x1cdb548\) null \[3\] 
&nbsp;            SVOP \(0x1cdb5c8\) const  PV \(0x1cd26c0\) "\\n" 
&nbsp;            SVOP \(0x1cdb490\) gv  GV \(0x1cd26a8\) \*\<none>&#8203;:: 
\-e syntax OK
\#\#\#\#\#

Marking ticket Resolved\.

Thank you very much\.
Jim Keenan

**Comment by p5pRT** [comment]
@jkeenan - Status changed from 'open' to 'resolved'
